ipam/ipam.go

```diff
@@ -79,15 +79,21 @@ func (ipam *IPAM) allocateIP(input interface{}, ctx common.RestContext) (interfa
 
 	// check for missing/empty required parameters
 	if tenantParam == "" {
-		return nil, errors.New("Missing or empty tenantName/tenantID parameter")
+		err := errors.New("Missing or empty tenantName/tenantID parameter")
+		log.Printf("IPAM encountered an error: %v", err)
```

Comment:
Why log it here instead of the caller who has (or should) to log and handle the error anyway?